x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/common/notifications/Level.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.common.notifications;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public enum Level implements Writeable {
+    INFO, ACTIVITY, WARNING, ERROR;
+
+    /**
+     * Case-insensitive from string method.
+     *
+     * @param value
+     *            String representation
+     * @return The condition type
+     */
+    public static Level fromString(String value) {
+        return Level.valueOf(value.toUpperCase(Locale.ROOT));
+    }
+
+    public static Level readFromStream(StreamInput in) throws IOException {
+        return in.readEnum(Level.class);
```

Comment:
I suppose we can't change this, just saying: would be better to map enum values explicit to e.g. a byte.

Reply:
This is how enums are serialised elsewhere in the Elasticsearch codebase.  I think we should fit into the pattern.  But for this particular class maybe we don't need to support wire serialisation at all.